redaxo/src/addons/media_manager/lib/effects/effect_header.php

```diff
@@ -10,6 +10,38 @@ public function execute()
         if ($this->params['cache'] == 'no_cache') {
             $this->media->setHeader('Cache-Control', 'must-revalidate, proxy-revalidate, private, no-cache, max-age=0');
             $this->media->setHeader('Expires', 'Sat, 26 Jul 1997 05:00:00 GMT'); // in the past
+        } elseif ($this->params['cache'] !== 'cache' /* bc */ && $this->params['cache'] !== 'unspecified') {
+            switch ($this->params['cache']) {
+                case '1 min':
+                    $seconds = 60;
+                    break;
+                case '1 hour':
+                    $seconds = 60 * 60;
+                    break;
+                case '1 day':
+                    $seconds = 60 * 60 * 24;
+                    break;
+                case '1 week':
+                    $seconds = 60 * 60 * 24 * 7;
+                    break;
+                case '1 month':
+                    $seconds = 60 * 60 * 24 * 30;
+                    break;
+                case '1 year':
+                case 'immutable':
+                    $seconds = 60 * 60 * 24 * 365;
+                    break;
+                default:
+                    throw new LogicException(sprintf('Unsupported cache duration "%s".', $this->params['cache']));
+            }
+
+            $cacheControl = 'proxy-revalidate, private, max-age='.$seconds;
```

Comment:
Sollte Proxy-Cache erlaubt sein, oder nicht?

Reply:
hmm ich würde entweder ein separates setting machen für proxy cache, oder den proxy cache (was letztendlich erfordert dass die leute sich mit den http headern auskennen) oder ggf. proxy-cache bei immutable ja, sonst immer revalidate? (vorschlag)

Reply:
> ggf. proxy-cache bei immutable ja, sonst immer revalidate?

Entweder man will bewusst, dass die Bilder nicht von Proxys gecacht werden, also nur direkt vom Browser, oder auch Proxys dürfen cachen. Aber die Frage ist für mich unabhängig davon, ob immutable oder nicht.
Hmm, vielleicht wirklich noch ne Einstellung dafür.

Reply:
der einzige use-case in dem es sinnvoll wäre dass man will das nur der browser cached, aber nicht der proxy ist in meinen augen wenn die bilder sensitive oder benutzer abhängige inhalte enthalten.

Reply:
Ja, den Use Case habe ich auch im Blick. Zum Beispiel in YCom oder so.
Und meine erste Idee war, hier pragmatisch einfach immer den Proxy-Cache abzuschalten, damit die Entwickler sich um dieses Thema keine Gedanken machen müssen.
(Was ja trotzdem aktiv ist, auch für Proxies, ist das Last-Modified/If-Modified-Since Verhalten).

Aber wie gesagt, bin hier offen, können dafür auch noch ne Einstellung einbauen.

Reply:
ich würds erstmal ohne einstellung machen. proxy-revalidate (ist wie alle anderen revalidate strategien der default, wenn ich es richtig verstehe)